client/tailscale/localclient.go

```diff
@@ -89,7 +89,7 @@ func (lc *LocalClient) socket() string {
 	if lc.Socket != "" {
 		return lc.Socket
 	}
-	return paths.DefaultTailscaledSocket()
+	return cmp.Or(envknob.String("TS_SOCKET"), paths.DefaultTailscaledSocket())
 }
```

Comment:
I worry that this might be a little too aggressive, not letting tsnet/LocalClient-using programs be in control without thus overriding them.

(Also it's not documented anywhere in those docs.)

If you only care about the `cmd/tailscale` binary, how about this:

```patch
diff --git a/cmd/tailscale/cli/cli.go b/cmd/tailscale/cli/cli.go
index 66961b2e0..eeb71509f 100644
--- a/cmd/tailscale/cli/cli.go
+++ b/cmd/tailscale/cli/cli.go
@@ -6,6 +6,7 @@
 package cli
 
 import (
+	"cmp"
 	"context"
 	"errors"
 	"flag"
@@ -165,12 +166,12 @@ func Run(args []string) (err error) {
 
 func newRootCmd() *ffcli.Command {
 	rootfs := newFlagSet("tailscale")
-	rootfs.Func("socket", "path to tailscaled socket", func(s string) error {
+	rootfs.Func("socket", "path to tailscaled socket; defaults to the TS_SOCKET environment value if set, else it uses the system default", func(s string) error {
 		localClient.Socket = s
 		localClient.UseSocketOnly = true
 		return nil
 	})
-	rootfs.Lookup("socket").DefValue = localClient.Socket
+	rootfs.Lookup("socket").DefValue = cmp.Or(envknob.String("TS_SOCKET"), localClient.Socket)
 
 	rootCmd := &ffcli.Command{
 		Name:       "tailscale",
```